packages/opentelemetry-runtime-metrics/src/metric.ts

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as api from '@opentelemetry/api';
+import {
+  HostMetrics,
+  ValueObserverWithObservations,
+} from '@opentelemetry/host-metrics';
+import * as enums from './enum';
+
+import { getHeapData, getMemoryData, getProcessData } from './stats/common';
+import { getStats } from './stats/native';
+
+import * as types from './types';
+
+/**
+ * Metrics Collector - collects metrics for CPU, Memory, Heap, Network, Event
+ * Loop, Garbage Collector, Heap Space
+ * the default label name for metric name is "name"
+ */
+export class RuntimeMetrics extends HostMetrics {
```

Comment:
At first glance, it looks like we'll inherit the recording of HostMetrics, but after a closer look this doesn't appear to be the case, since we override the `_createMetrics()` function. Would it make more sense to have a common base class with shared functionality that HostMetrics and Runtime metrics can subclass? For example, `class RuntimeMetrics extends MetricsBase {}` and `class HostMetrics extends MetricsBase {}`.

Reply:
done

Reply:
Thanks @obecny 👀 great!